eng/common/pipelines/templates/steps/verify-resource-ref.yml

```diff
@@ -0,0 +1,32 @@
+
+steps:
+  - pwsh: |
+      Install-Module -Name powershell-yaml -RequiredVersion 0.4.1 -Force -Scope CurrentUser
+      $remoteHEAD = git rev-parse origin/master
+      $localHEAD = git rev-parse HEAD
+      $ymlfilesChanged = (git diff --name-only $remoteHEAD...$localHEAD) | Where-Object {$_ -like '*.yml'}
+      
+      foreach ($file in $ymlfilesChanged)
+      {
+        Write-Host "Verifying "$file
+        $ymlContent = Get-Content "$(Build.SourcesDirectory)/$file" -Raw
+        $ymlObject = ConvertFrom-Yaml $ymlContent -Ordered
+
+        if ($ymlObject.Contains("resources"))
```

Comment:
Is the contains check necessary? Won't PS simply return you a null or empty list if `$ymlObject["resources"]["repositories"]` doesn't have anything?

Reply:
If the first case is none existent it will throw `InvalidOperation: Cannot index into a null array.`